src/node_crypto.cc

```diff
@@ -638,6 +652,16 @@ void SecureContext::Init(const FunctionCallbackInfo<Value>& args) {
   SSL_CTX_set_options(sc->ctx_, SSL_OP_NO_SSLv2);
   SSL_CTX_set_options(sc->ctx_, SSL_OP_NO_SSLv3);
 
+  // If TLS 1.3, disable All older tls protocols
```

Comment:
Why?

Reply:
there is no "TLSv1_3_client_method" or "TLSv1_3_server_method" in the OpenSSL 1.1.1 API. Instead, you use SSL_CTX_set_min_proto_version(sc->ctx_, TLS1_3_VERSION) to make it use 1.3 as the "lowest" protocol when you need TLS 1.3 specifically.

Reply:
FWIW it seems to at least be in OpenSSL's master branch

Reply:
From Matt Caswell:

orry for the slightly slow response on this. I am away this week travelling (https://www.openssl.org/blog/blog/2017/08/28/china/). 

Some comments inserted below. 

On Mon, 18 Sep 2017 07:25:18 -0700 
Mahesh Bhoothapuri <[hidden email]> wrote: 

> Thanks for responding.  Yes,  I have done the steps mentioned above.   Here 
> are my settings: 
> 
>     int min_version = TLS1_3_VERSION, max_version = TLS1_3_VERSION; 
> 
>     meth = isClient ? tlsv1_3_client_method() : tlsv1_3_server_method(); 

This is *not* correct. These are internal functions that are not exposed via the public header files or exported by the library. You should not be using them at all.

Reply:
http://openssl.6102.n7.nabble.com/TLS-1-3-client-hello-issue-tp72449p72460.html